tests/Doctrine/Tests/DBAL/Platforms/SQLServer2012PlatformTest.php

```diff
@@ -371,4 +372,20 @@ public function testModifyLimitQueryWithNewlineBeforeOrderBy()
         $sql = $this->_platform->modifyLimitQuery($querySql, 10);
         $this->assertEquals($expectedSql, $sql);
     }
+
+    public function testGetDefaultValueDeclarationSQLForDateType()
+    {
+        $currentDateSql = $this->_platform->getCurrentDateSQL();
+	    foreach (array('date', 'date_immutable') as $type) {
+		    $field = array(
+			    'type'    => Type::getType($type),
+			    'default' => $currentDateSql,
+		    );
+		
+		    $this->assertEquals(
```

Comment:
Please use `self::assert*` instead, and it would be better to use `assertSame()` in this case